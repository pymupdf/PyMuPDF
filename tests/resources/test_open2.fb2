<?xml version="1.0" encoding="UTF-8"?>
<FictionBook xmlns="http://www.gribuser.ru/xml/fictionbook/2.0" xmlns:xlink="http://www.w3.org/1999/xlink">
<description>
    <title-info>
        <genre>computers</genre>
        <author>
            <first-name>Chris</first-name>
            <last-name>Clark</last-name>
        </author>
        <book-title>Sample FB2 book</book-title>
        <annotation>
            <p>Short sample of a FictionBook2 book with simple metadata. Based on test_book.md from https://github.com/clach04/sample_reading_media</p>
        </annotation>
        <keywords>ebook,sample,markdown,fb2,FictionBook2</keywords>
    </title-info>
    <document-info>
        <author>
            <nickname>clach04</nickname>
            <home-page>https://github.com/clach04/sample_reading_media</home-page>
        </author>

        <program-used>vim and scite</program-used>
        <src-url>https://github.com/clach04/sample_reading_media</src-url>
        <version>1.0</version>
        <history>
            <p>Initial version, written by hand.</p>
        </history>
    </document-info>
</description>
<body>
    <title>
        <p>This is a title</p>
    </title>

    <section id="test-header-h1">
        <title>
            <p>Test Header h1</p>
        </title>

        <p>A test paragraph.</p>
        <p>Another test paragraph.</p>
    </section>

    <section id="another-test-header-h1">
        <title>
            <p>Another Test Header h1</p>
        </title>

        <section id="a-test-header-h2">
            <title>
                <p>A Test Header h2</p>
            </title>
        
            <section id="a-test-header-h3">
                <title>
                    <p>A Test Header h3</p>
                </title>

                <p>Yet more copy</p>
            </section>
        </section>
    </section>
</body>
</FictionBook>
