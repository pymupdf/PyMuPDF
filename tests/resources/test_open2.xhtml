
<?xml version="1.0"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<style>
body{background-color:gray}
div{background-color:white;margin:1em;padding:1em}
p{white-space:pre-wrap}
</style>
</head>
<body>
<div id="page0">
<p>Some text</p>
</div>
</body>
</html>
