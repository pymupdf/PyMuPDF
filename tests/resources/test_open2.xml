
<?xml version="1.0"?>
<document name="xml-file.xml">
<page id="page0" width="612" height="792">
<block bbox="100 88.175 149.52199 103.289" justify="unknown">
<line bbox="100 88.175 149.52199 103.289" wmode="0" dir="1 0" text="Some text">
<font name="Helvetica" size="11">
<char quad="100 88.175 107.337 88.175 100 103.289 107.337 103.289" x="100" y="100" bidi="0" color="#000000" alpha="#ff" flags="16" c="S"/>
<char quad="107.337 88.175 113.452999 88.175 107.337 103.289 113.452999 103.289" x="107.337" y="100" bidi="0" color="#000000" alpha="#ff" flags="16" c="o"/>
<char quad="113.452999 88.175 122.616 88.175 113.452999 103.289 122.616 103.289" x="113.452999" y="100" bidi="0" color="#000000" alpha="#ff" flags="16" c="m"/>
<char quad="122.616 88.175 128.732 88.175 122.616 103.289 128.732 103.289" x="122.616" y="100" bidi="0" color="#000000" alpha="#ff" flags="16" c="e"/>
<char quad="128.732 88.175 131.79 88.175 128.732 103.289 131.79 103.289" x="128.732" y="100" bidi="0" color="#000000" alpha="#ff" flags="16" c=" "/>
<char quad="131.79 88.175 134.84799 88.175 131.79 103.289 134.84799 103.289" x="131.79" y="100" bidi="0" color="#000000" alpha="#ff" flags="16" c="t"/>
<char quad="134.84799 88.175 140.96399 88.175 134.84799 103.289 140.96399 103.289" x="134.84799" y="100" bidi="0" color="#000000" alpha="#ff" flags="16" c="e"/>
<char quad="140.96399 88.175 146.46399 88.175 140.96399 103.289 146.46399 103.289" x="140.96399" y="100" bidi="0" color="#000000" alpha="#ff" flags="16" c="x"/>
<char quad="146.46399 88.175 149.52199 88.175 146.46399 103.289 149.52199 103.289" x="146.46399" y="100" bidi="0" color="#000000" alpha="#ff" flags="16" c="t"/>
</font>
</line>
</block>
</page>
</document>
