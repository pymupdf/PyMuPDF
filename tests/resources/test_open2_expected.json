{
    "tests/resources/test_open2.cbz": {
        "": {
            "file": "zip",
            "stream": "zip"
        },
        ".cbz": {
            "file": "zip",
            "stream": "zip"
        },
        ".doc": {
            "file": "zip",
            "stream": "zip"
        },
        ".docx": {
            "file": "zip",
            "stream": "zip"
        },
        ".epub": {
            "file": "zip",
            "stream": "zip"
        },
        ".fb2": {
            "file": "zip",
            "stream": "zip"
        },
        ".html": {
            "file": "zip",
            "stream": "zip"
        },
        ".jpg": {
            "file": "zip",
            "stream": "zip"
        },
        ".mobi": {
            "file": "zip",
            "stream": "zip"
        },
        ".pdf": {
            "file": "zip",
            "stream": "zip"
        },
        ".svg": {
            "file": "zip",
            "stream": "zip"
        },
        ".txt": {
            "file": "zip",
            "stream": "zip"
        },
        ".xhtml": {
            "file": "zip",
            "stream": "zip"
        },
        ".xml": {
            "file": "zip",
            "stream": "zip"
        },
        ".xps": {
            "file": "zip",
            "stream": "zip"
        }
    },
    "tests/resources/test_open2.doc": {
        "": {
            "file": "[error]",
            "stream": "[error]"
        },
        ".cbz": {
            "file": "cfb",
            "stream": "cfb"
        },
        ".doc": {
            "file": "[error]",
            "stream": "[error]"
        },
        ".docx": {
            "file": "[error]",
            "stream": "[error]"
        },
        ".epub": {
            "file": "[error]",
            "stream": "[error]"
        },
        ".fb2": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        },
        ".html": {
            "file": "HTML5",
            "stream": "HTML5"
        },
        ".jpg": {
            "file": "Image",
            "stream": "Image"
        },
        ".mobi": {
            "file": "[error]",
            "stream": "[error]"
        },
        ".pdf": {
            "file": "[error]",
            "stream": "[error]"
        },
        ".svg": {
            "file": "SVG",
            "stream": "SVG"
        },
        ".txt": {
            "file": "Text",
            "stream": "Text"
        },
        ".xhtml": {
            "file": "XHTML",
            "stream": "XHTML"
        },
        ".xml": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        },
        ".xps": {
            "file": "[error]",
            "stream": "[error]"
        }
    },
    "tests/resources/test_open2.docx": {
        "": {
            "file": "Office document",
            "stream": "Office document"
        },
        ".cbz": {
            "file": "Office document",
            "stream": "Office document"
        },
        ".doc": {
            "file": "Office document",
            "stream": "Office document"
        },
        ".docx": {
            "file": "Office document",
            "stream": "Office document"
        },
        ".epub": {
            "file": "Office document",
            "stream": "Office document"
        },
        ".fb2": {
            "file": "Office document",
            "stream": "Office document"
        },
        ".html": {
            "file": "Office document",
            "stream": "Office document"
        },
        ".jpg": {
            "file": "Office document",
            "stream": "Office document"
        },
        ".mobi": {
            "file": "Office document",
            "stream": "Office document"
        },
        ".pdf": {
            "file": "Office document",
            "stream": "Office document"
        },
        ".svg": {
            "file": "Office document",
            "stream": "Office document"
        },
        ".txt": {
            "file": "Office document",
            "stream": "Office document"
        },
        ".xhtml": {
            "file": "Office document",
            "stream": "Office document"
        },
        ".xml": {
            "file": "Office document",
            "stream": "Office document"
        },
        ".xps": {
            "file": "Office document",
            "stream": "Office document"
        }
    },
    "tests/resources/test_open2.epub": {
        "": {
            "file": "EPUB",
            "stream": "EPUB"
        },
        ".cbz": {
            "file": "zip",
            "stream": "zip"
        },
        ".doc": {
            "file": "EPUB",
            "stream": "EPUB"
        },
        ".docx": {
            "file": "EPUB",
            "stream": "EPUB"
        },
        ".epub": {
            "file": "EPUB",
            "stream": "EPUB"
        },
        ".fb2": {
            "file": "EPUB",
            "stream": "EPUB"
        },
        ".html": {
            "file": "EPUB",
            "stream": "EPUB"
        },
        ".jpg": {
            "file": "EPUB",
            "stream": "EPUB"
        },
        ".mobi": {
            "file": "EPUB",
            "stream": "EPUB"
        },
        ".pdf": {
            "file": "EPUB",
            "stream": "EPUB"
        },
        ".svg": {
            "file": "EPUB",
            "stream": "EPUB"
        },
        ".txt": {
            "file": "EPUB",
            "stream": "EPUB"
        },
        ".xhtml": {
            "file": "EPUB",
            "stream": "EPUB"
        },
        ".xml": {
            "file": "EPUB",
            "stream": "EPUB"
        },
        ".xps": {
            "file": "EPUB",
            "stream": "EPUB"
        }
    },
    "tests/resources/test_open2.fb2": {
        "": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        },
        ".cbz": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        },
        ".doc": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        },
        ".docx": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        },
        ".epub": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        },
        ".fb2": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        },
        ".html": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        },
        ".jpg": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        },
        ".mobi": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        },
        ".pdf": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        },
        ".svg": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        },
        ".txt": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        },
        ".xhtml": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        },
        ".xml": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        },
        ".xps": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        }
    },
    "tests/resources/test_open2.html": {
        "": {
            "file": "HTML5",
            "stream": "HTML5"
        },
        ".cbz": {
            "file": "HTML5",
            "stream": "HTML5"
        },
        ".doc": {
            "file": "HTML5",
            "stream": "HTML5"
        },
        ".docx": {
            "file": "HTML5",
            "stream": "HTML5"
        },
        ".epub": {
            "file": "HTML5",
            "stream": "HTML5"
        },
        ".fb2": {
            "file": "HTML5",
            "stream": "HTML5"
        },
        ".html": {
            "file": "HTML5",
            "stream": "HTML5"
        },
        ".jpg": {
            "file": "HTML5",
            "stream": "HTML5"
        },
        ".mobi": {
            "file": "HTML5",
            "stream": "HTML5"
        },
        ".pdf": {
            "file": "HTML5",
            "stream": "HTML5"
        },
        ".svg": {
            "file": "HTML5",
            "stream": "HTML5"
        },
        ".txt": {
            "file": "HTML5",
            "stream": "HTML5"
        },
        ".xhtml": {
            "file": "XHTML",
            "stream": "XHTML"
        },
        ".xml": {
            "file": "HTML5",
            "stream": "HTML5"
        },
        ".xps": {
            "file": "HTML5",
            "stream": "HTML5"
        }
    },
    "tests/resources/test_open2.jpg": {
        "": {
            "file": "Image",
            "stream": "Image"
        },
        ".cbz": {
            "file": "Image",
            "stream": "Image"
        },
        ".doc": {
            "file": "Image",
            "stream": "Image"
        },
        ".docx": {
            "file": "Image",
            "stream": "Image"
        },
        ".epub": {
            "file": "Image",
            "stream": "Image"
        },
        ".fb2": {
            "file": "Image",
            "stream": "Image"
        },
        ".html": {
            "file": "Image",
            "stream": "Image"
        },
        ".jpg": {
            "file": "Image",
            "stream": "Image"
        },
        ".mobi": {
            "file": "Image",
            "stream": "Image"
        },
        ".pdf": {
            "file": "Image",
            "stream": "Image"
        },
        ".svg": {
            "file": "Image",
            "stream": "Image"
        },
        ".txt": {
            "file": "Image",
            "stream": "Image"
        },
        ".xhtml": {
            "file": "Image",
            "stream": "Image"
        },
        ".xml": {
            "file": "Image",
            "stream": "Image"
        },
        ".xps": {
            "file": "Image",
            "stream": "Image"
        }
    },
    "tests/resources/test_open2.mobi": {
        "": {
            "file": "MOBI",
            "stream": "MOBI"
        },
        ".cbz": {
            "file": "MOBI",
            "stream": "MOBI"
        },
        ".doc": {
            "file": "MOBI",
            "stream": "MOBI"
        },
        ".docx": {
            "file": "MOBI",
            "stream": "MOBI"
        },
        ".epub": {
            "file": "MOBI",
            "stream": "MOBI"
        },
        ".fb2": {
            "file": "MOBI",
            "stream": "MOBI"
        },
        ".html": {
            "file": "MOBI",
            "stream": "MOBI"
        },
        ".jpg": {
            "file": "MOBI",
            "stream": "MOBI"
        },
        ".mobi": {
            "file": "MOBI",
            "stream": "MOBI"
        },
        ".pdf": {
            "file": "MOBI",
            "stream": "MOBI"
        },
        ".svg": {
            "file": "MOBI",
            "stream": "MOBI"
        },
        ".txt": {
            "file": "MOBI",
            "stream": "MOBI"
        },
        ".xhtml": {
            "file": "MOBI",
            "stream": "MOBI"
        },
        ".xml": {
            "file": "MOBI",
            "stream": "MOBI"
        },
        ".xps": {
            "file": "MOBI",
            "stream": "MOBI"
        }
    },
    "tests/resources/test_open2.pdf": {
        "": {
            "file": "PDF 1.5",
            "stream": "PDF 1.5"
        },
        ".cbz": {
            "file": "PDF 1.5",
            "stream": "PDF 1.5"
        },
        ".doc": {
            "file": "PDF 1.5",
            "stream": "PDF 1.5"
        },
        ".docx": {
            "file": "PDF 1.5",
            "stream": "PDF 1.5"
        },
        ".epub": {
            "file": "PDF 1.5",
            "stream": "PDF 1.5"
        },
        ".fb2": {
            "file": "PDF 1.5",
            "stream": "PDF 1.5"
        },
        ".html": {
            "file": "PDF 1.5",
            "stream": "PDF 1.5"
        },
        ".jpg": {
            "file": "PDF 1.5",
            "stream": "PDF 1.5"
        },
        ".mobi": {
            "file": "PDF 1.5",
            "stream": "PDF 1.5"
        },
        ".pdf": {
            "file": "PDF 1.5",
            "stream": "PDF 1.5"
        },
        ".svg": {
            "file": "PDF 1.5",
            "stream": "PDF 1.5"
        },
        ".txt": {
            "file": "PDF 1.5",
            "stream": "PDF 1.5"
        },
        ".xhtml": {
            "file": "PDF 1.5",
            "stream": "PDF 1.5"
        },
        ".xml": {
            "file": "PDF 1.5",
            "stream": "PDF 1.5"
        },
        ".xps": {
            "file": "PDF 1.5",
            "stream": "PDF 1.5"
        }
    },
    "tests/resources/test_open2.svg": {
        "": {
            "file": "SVG",
            "stream": "SVG"
        },
        ".cbz": {
            "file": "SVG",
            "stream": "SVG"
        },
        ".doc": {
            "file": "SVG",
            "stream": "SVG"
        },
        ".docx": {
            "file": "SVG",
            "stream": "SVG"
        },
        ".epub": {
            "file": "SVG",
            "stream": "SVG"
        },
        ".fb2": {
            "file": "SVG",
            "stream": "SVG"
        },
        ".html": {
            "file": "SVG",
            "stream": "SVG"
        },
        ".jpg": {
            "file": "SVG",
            "stream": "SVG"
        },
        ".mobi": {
            "file": "SVG",
            "stream": "SVG"
        },
        ".pdf": {
            "file": "SVG",
            "stream": "SVG"
        },
        ".svg": {
            "file": "SVG",
            "stream": "SVG"
        },
        ".txt": {
            "file": "SVG",
            "stream": "SVG"
        },
        ".xhtml": {
            "file": "SVG",
            "stream": "SVG"
        },
        ".xml": {
            "file": "SVG",
            "stream": "SVG"
        },
        ".xps": {
            "file": "SVG",
            "stream": "SVG"
        }
    },
    "tests/resources/test_open2.xhtml": {
        "": {
            "file": "XHTML",
            "stream": "XHTML"
        },
        ".cbz": {
            "file": "XHTML",
            "stream": "XHTML"
        },
        ".doc": {
            "file": "XHTML",
            "stream": "XHTML"
        },
        ".docx": {
            "file": "XHTML",
            "stream": "XHTML"
        },
        ".epub": {
            "file": "XHTML",
            "stream": "XHTML"
        },
        ".fb2": {
            "file": "XHTML",
            "stream": "XHTML"
        },
        ".html": {
            "file": "HTML5",
            "stream": "HTML5"
        },
        ".jpg": {
            "file": "XHTML",
            "stream": "XHTML"
        },
        ".mobi": {
            "file": "XHTML",
            "stream": "XHTML"
        },
        ".pdf": {
            "file": "XHTML",
            "stream": "XHTML"
        },
        ".svg": {
            "file": "XHTML",
            "stream": "XHTML"
        },
        ".txt": {
            "file": "XHTML",
            "stream": "XHTML"
        },
        ".xhtml": {
            "file": "XHTML",
            "stream": "XHTML"
        },
        ".xml": {
            "file": "XHTML",
            "stream": "XHTML"
        },
        ".xps": {
            "file": "XHTML",
            "stream": "XHTML"
        }
    },
    "tests/resources/test_open2.xml": {
        "": {
            "file": "[error]",
            "stream": "[error]"
        },
        ".cbz": {
            "file": "[error]",
            "stream": "[error]"
        },
        ".doc": {
            "file": "[error]",
            "stream": "[error]"
        },
        ".docx": {
            "file": "[error]",
            "stream": "[error]"
        },
        ".epub": {
            "file": "[error]",
            "stream": "[error]"
        },
        ".fb2": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        },
        ".html": {
            "file": "HTML5",
            "stream": "HTML5"
        },
        ".jpg": {
            "file": "Image",
            "stream": "Image"
        },
        ".mobi": {
            "file": "[error]",
            "stream": "[error]"
        },
        ".pdf": {
            "file": "[error]",
            "stream": "[error]"
        },
        ".svg": {
            "file": "SVG",
            "stream": "SVG"
        },
        ".txt": {
            "file": "Text",
            "stream": "Text"
        },
        ".xhtml": {
            "file": "XHTML",
            "stream": "XHTML"
        },
        ".xml": {
            "file": "FictionBook2",
            "stream": "FictionBook2"
        },
        ".xps": {
            "file": "[error]",
            "stream": "[error]"
        }
    },
    "tests/resources/test_open2.xps": {
        "": {
            "file": "XPS",
            "stream": "XPS"
        },
        ".cbz": {
            "file": "zip",
            "stream": "zip"
        },
        ".doc": {
            "file": "XPS",
            "stream": "XPS"
        },
        ".docx": {
            "file": "XPS",
            "stream": "XPS"
        },
        ".epub": {
            "file": "XPS",
            "stream": "XPS"
        },
        ".fb2": {
            "file": "XPS",
            "stream": "XPS"
        },
        ".html": {
            "file": "XPS",
            "stream": "XPS"
        },
        ".jpg": {
            "file": "XPS",
            "stream": "XPS"
        },
        ".mobi": {
            "file": "XPS",
            "stream": "XPS"
        },
        ".pdf": {
            "file": "XPS",
            "stream": "XPS"
        },
        ".svg": {
            "file": "XPS",
            "stream": "XPS"
        },
        ".txt": {
            "file": "XPS",
            "stream": "XPS"
        },
        ".xhtml": {
            "file": "XPS",
            "stream": "XPS"
        },
        ".xml": {
            "file": "XPS",
            "stream": "XPS"
        },
        ".xps": {
            "file": "XPS",
            "stream": "XPS"
        }
    }
}